go/adcm/status/host.go

```diff
@@ -0,0 +1,169 @@
+package status
+
+type Host struct {
+	Id              int  `json:"id"`
+	MaintenanceMode bool `json:"maintenance_mode"`
+}
+
+type dbHost map[int]Host
+
+type hostRequest struct {
+	command          string
+	id               int
+	maintenance_mode bool
+}
+
+type hostResponse struct {
+	code int
+	ok   bool
+	host Host
+}
+
+type HostStorage struct {
+	in    chan hostRequest
+	out   chan hostResponse
+	db    dbHost
+	label string
+}
+
+func newHostStorage(db dbHost, label string) *HostStorage {
+	return &HostStorage{
+		in:    make(chan hostRequest),
+		out:   make(chan hostResponse),
+		db:    db,
+		label: label,
+	}
+}
+
+func (hs *HostStorage) run() {
+	logg.I.f("start storage %s server", hs.label)
+	for {
+		request := <-hs.in
+		logg.I.f("Storage %s command: %+v", hs.label, request.command)
+		switch request.command {
+		case "retrieve":
+			value, ok := hs.db.retrieve(request.id)
+			hs.out <- hostResponse{host: value, ok: ok}
+		case "update":
+			value := hs.db.update(request.id, request.maintenance_mode)
+			hs.out <- hostResponse{code: value}
+		default:
+			logg.E.f("Storage %s unknown command: %+v", hs.label, request)
+
+		}
+	}
+}
+
+func (hs *HostStorage) retrieve(id int) (Host, bool) {
+	request := hostRequest{command: "retrieve", id: id}
+	hs.in <- request
+	response := <-hs.out
+	return response.host, response.ok
+}
+
+func (hs *HostStorage) update(id int, maintenance_mode bool) int {
+	request := hostRequest{command: "update", id: id, maintenance_mode: maintenance_mode}
+	hs.in <- request
+	response := <-hs.out
+	return response.code
+}
+
+func (db dbHost) retrieve(id int) (Host, bool) {
+	value, ok := db[id]
+	if ok {
+		return value, true
+	}
+	return Host{Id: id, MaintenanceMode: false}, false
+}
+
+func (db dbHost) update(id int, maintenance_mode bool) int {
+	code := 200
+	host, ok := db[id]
+	if !ok {
+		db[id] = Host{Id: id, MaintenanceMode: maintenance_mode}
+		code = 201
+	}
```

Comment:
it looks like `update_or_create`, but maybe something like `setHostInfo` with a little docstring?